internal/datastore/crdb/README.md

```diff
@@ -8,3 +8,103 @@ The CockroachDB implementation should be used when your SpiceDB service runs in
+
+Now we know when timestamps _could_ be reversed. But when does that matter to your application?
+
+The TL;DR is: only when you care about the New Enemy Problem.
+
+Let's take a look at a couple of examples and show how it could be an issue for an application storing permissions in SpiceDB.
+
+### Neglecting ACL Update Order
+
+Two separate `WriteRelationship` calls come in:
+
+`A`: Alice removes Bob from the `shared` folder
+
+`B`: Alice adds a new document `not-for-bob.txt` to the `shared` folder
+
+The normal case is that the timestamp for `A` < the timestamp for `B`.
+
+But if those two writes hit the conditions for a timestamp reversal, then `B < A`.
+
+From Alice's perspective, there should be no time at which Bob can ever see `not-for-bob.txt`.
+She performed the first write, got a response, and then performed the second write.
+
+But this isn't true when using `MinimizeLatency` or `AtLeastAsFresh` consistency.
+If Bob later performs a `Check` request for the `not-for-bob.txt` document, it's possible that SpiceDB will pick an evaluation timestamp `T` such that `B < T < A`, so that the document is in the folder _and_ bob is allowed to see the contents of the folder.
+
+Note that this is only possible if `A - T < quantization window`: the check has to happen soon enough after the write for `A` that it's possible that SpiceDB picks a timestamp in between them.
+The default quantization window is `5s`.
+
+### Application Mitigations for ACL Update Order
+
+This could be mitigated in your application by:
+
+- Not caring about the problem
+- Not allowing the write from `B` within the max_offset time of the CRDB cluster (or the quantization window).
+- Not allowing a Check on a resource within max_offset of its ACL modification (or the quantization window).
+
+## Mis-apply Old ACLs to New Content
+
+Two separate API calls come in:
+
+`A`: Alice removes Bob as a viewer of document `secret`
+
+`B`: Alice does a `FullyConsistent` `Check` request to get a ZedToken
+
+`C`: Alice stores that ZedToken (timestamp `B`) with the document `secret` when she updates it to say `Bob is a fool`.
+
+Same as before, the normal case is that the timestamp for `A` < the timestamp for `B`, but if the two writes hit the conditions for a timestamp reversal, then `B < A`.
+
+Bob later tries to read the document. The application performs an `AtLeastAsFresh` `Check` for Bob to access the document `secret` using the stored Zedtoken (which is timestamp `B`.)
+
+It's possible that SpiceDB will pick an evaluation timestamp `T` such that `B < T < A`, so that bob is allowed to read the newest contents of the document, and discover that Alice thinks he is a fool.
+
+Same as before, this is only possible if `A - T < quantization window`: Bob's check has to happen soon enough after the write for `A` that it's possible that SpiceDB picks a timestamp in between `A` and `B`, and the default quantization window is `5s`.
+
+### Application Mitigations for Misapplying Old ACLs
```

Comment:
nit: may be better to make this a subsection of `Mis-apply Old ACLs to New Content`

```suggestion
#### Application Mitigations for Misapplying Old ACLs
```